File: code/PrefixtoPostFix.java
import java.util.Stack;
public class PrefixtoPostFix {

    boolean isOperator(char x){
        switch (x){
            case '-':
            case '+':
            case '/':
            case '*':
            case '^':
            return true;
        }
        return false;
    }
    
    public String convert(String expression){

        Stack<String> stack = new Stack<String>();
        for (int i = expression.length()-1; i >=0 ; i--) {

            char c = expression.charAt(i);

            if(isOperator(c)){
                String s1 = stack.pop();
                String s2 = stack.pop();
                String temp = s1 + s2 + c;
                stack.push(temp);
            }else{
                stack.push(c+"");
            }
        }

        String result = stack.pop();
        return result;
    }

    public static void main(String[] args) {
        String prefix = "*/93+*24-76";
        System.out.println("Prefix Expression: " + prefix);
        System.out.println("Postfix Expression: " + new PrefixtoPostFix().convert(prefix));
    }
}

File: code/PrefixtoPostFix.py
class PrefixtoPostFix:
    def isOperator(self, x):
        return x in {'-', '+', '/', '*', '^'}

    def convert(self, expression):
        stack = []
        for i in range(len(expression)-1, -1, -1):
            c = expression[i]
            if self.isOperator(c):
                s1 = stack.pop()
                s2 = stack.pop()
                temp = s1 + s2 + c
                stack.append(temp)
            else:
                stack.append(c)
        result = stack.pop()
        return result


if __name__ == "__main__":
    prefix = "*/93+*24-76"
    print("Prefix Expression:", prefix)
    print("Postfix Expression:", PrefixtoPostFix().convert(prefix))
